Limelight/Stream/Connection.m

```diff
@@ -310,63 +315,86 @@ -(void) terminate
     _serverInfo.serverInfoAppVersion = [config.appVersion cStringUsingEncoding:NSUTF8StringEncoding];
     if (config.gfeVersion != nil) {
         _serverInfo.serverInfoGfeVersion = [config.gfeVersion cStringUsingEncoding:NSUTF8StringEncoding];
     }
-    
+
     renderer = myRenderer;
     _callbacks = callbacks;
-    
+
     LiInitializeStreamConfiguration(&_streamConfig);
     _streamConfig.width = config.width;
     _streamConfig.height = config.height;
     _streamConfig.fps = config.frameRate;
     _streamConfig.bitrate = config.bitRate;
-    
+
+    //This will activate the remote streaming optimization in moonlight-common if needed
+    _streamConfig.streamingRemotely = config.streamingRemotely;
+
+#if TARGET_OS_IPHONE
     // On iOS 11, we can use HEVC if the server supports encoding it
     // and this device has hardware decode for it (A9 and later)
     if (@available(iOS 11.0, *)) {
         // FIXME: Disabled due to incompatibility with iPhone X causing video
         // to freeze. Additionally, RFI is not supported so packet loss recovery
         // is worse with HEVC than H.264.
+
+        //Streaming with a limited bandwith will result in better quality with HEVC
         //_streamConfig.supportsHevc = VTIsHardwareDecodeSupported(kCMVideoCodecType_HEVC);
     }
+#else
+    if (@available(macOS 10.13, *)) {
+        _streamConfig.supportsHevc = true;
+    }
+#endif
     
     // Use some of the HEVC encoding efficiency improvements to
     // reduce bandwidth usage while still gaining some image
     // quality improvement.
-    _streamConfig.hevcBitratePercentageMultiplier = 75;
-    
-    // FIXME: We should use 1024 when streaming remotely
-    _streamConfig.packetSize = 1292;
-    
+    if (config.streamingRemotely) {
+        //In the case of remotely streaming, we want the best possible qualtity for a limited bandwidth, so we set the multiplier to 0
+        _streamConfig.hevcBitratePercentageMultiplier = 0;
+        //When streaming remotely we want to use a packet size of 1024
+        _streamConfig.packetSize = 1024;
+    }
+    else {
+        _streamConfig.hevcBitratePercentageMultiplier = 75;
+        _streamConfig.packetSize = 1292;
+    }
+
     memcpy(_streamConfig.remoteInputAesKey, [config.riKey bytes], [config.riKey length]);
     memset(_streamConfig.remoteInputAesIv, 0, 16);
     int riKeyId = htonl(config.riKeyId);
     memcpy(_streamConfig.remoteInputAesIv, &riKeyId, sizeof(riKeyId));
-    
+
     LiInitializeVideoCallbacks(&_drCallbacks);
     _drCallbacks.setup = DrDecoderSetup;
     _drCallbacks.submitDecodeUnit = DrSubmitDecodeUnit;
-    
+
+
+#if TARGET_OS_IPHONE
     // RFI doesn't work properly with HEVC on iOS 11 with an iPhone SE (at least)
     _drCallbacks.capabilities = CAPABILITY_REFERENCE_FRAME_INVALIDATION_AVC;
-    
+#else
+    // It works on macOS (on my MBP atleast)
+    _drCallbacks.capabilities = CAPABILITY_REFERENCE_FRAME_INVALIDATION_HEVC;
```

Comment:
Did you test with packet loss scenarios?

You should also specify CAPABILITY_REFERENCE_FRAME_INVALIDATION_AVC here if relevant.

Reply:
I was wrong here, sorry. I assumed that a connection via Internet would loose some frames here and there, but it doesn't seem so. I have now tested it with the Network Link Conditioner and came to the conclusion that it actually doesn't work.